src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/InterchangeHeader.java

```diff
@@ -23,6 +24,7 @@
 
     public static final String KEY = "UNB";
     private static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("yyMMdd:HHmm").withZone(TimestampService.UKZone);
+    private static final Long MAX_INTERCHANGE_SEQUENCE = SequenceRepository.MAX_SEQUENCE_NUMBER - 1;
```

Comment:
`SequenceRepository.MAX_SEQUENCE_NUMBER - 1` is probably not a great idea and MAX_SEQUENCE_NUMBER should probably be private. What happens if the implementation changes?

Reply:
Shouldn't sequence number generator max value be in sync with allowed max in segment class?